guacamole/src/main/java/org/apache/guacamole/rest/APIRequest.java

```diff
@@ -58,6 +68,27 @@ public APIRequest(HttpServletRequest request,
 
         super(request);
 
+        // Try a few methods to get client info.
+        if(request.getHeader("X-Guacamole-Client-Hostname") != null && request.getHeader("X-Guacamole-Client-Hostname") != "") {
```

Comment:
Depending on how Guacamole + Tomcat is configured, namely behind a proxy, it can be difficult to get the original client information through.  By default, the remote host/IP info when Tomcat is behind an Apache httpd or nginx server is whatever the hostname and IP of the proxy is, rather than the client.  This is not the intended behavior of the module, since you want the hostname and/or IP of the actual client, not the proxy server.  I implemented it this way in order to give some options - you can set up custom headers that pass through, you can do some Apache+Tomcat configuration, or you can rely on the standard remote host/addr fields.  See comments on the JIRA issue: https://issues.apache.org/jira/browse/GUACAMOLE-47